src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancies/AssetOccupancyManager.cs

```diff
@@ -18,73 +17,53 @@ namespace EasyAbp.BookingService.AssetOccupancies;
-        DefaultPeriodSchemeStore defaultPeriodSchemeStore,
-        IAssetCategoryRepository assetCategoryRepository,
+        DefaultPeriodSchemeProvider defaultPeriodSchemeProvider,
         IAssetScheduleRepository assetScheduleRepository,
         IOptions<BookingServiceOptions> options)
     {
         _repository = repository;
-        _assetRepository = assetRepository;
         _periodSchemeRepository = periodSchemeRepository;
         _assetPeriodSchemeRepository = assetPeriodSchemeRepository;
-        _defaultPeriodSchemeStore = defaultPeriodSchemeStore;
-        _assetCategoryRepository = assetCategoryRepository;
+        _defaultPeriodSchemeProvider = defaultPeriodSchemeProvider;
         _assetScheduleRepository = assetScheduleRepository;
         _options = options.Value;
     }
 
     [UnitOfWork]
-    public virtual async Task<List<BookablePeriod>> SearchAssetBookablePeriodsAsync(Asset asset,
+    public virtual async Task<List<PeriodOccupancyModel>> SearchAssetBookablePeriodsAsync(Asset asset,
         AssetCategory category,
-        DateTime bookingDateTime,
-        DateTime searchDate)
+        DateTime currentDateTime,
+        DateTime targetDate)
     {
-        var assetPeriodScheme = await _assetPeriodSchemeRepository.FindAsync(x =>
-            x.AssetId == asset.Id && x.Date == searchDate);
-
-        var effectivePeriodScheme = await CalculateEffectivePeriodSchemeAsync(
-            assetPeriodScheme, asset, category);
-
-        if (effectivePeriodScheme.Periods.IsNullOrEmpty())
-        {
-            return new List<BookablePeriod>();
-        }
-
-        var endingTime = effectivePeriodScheme.Periods
-            .Select(x => x.GetEndingTime())
-            .OrderByDescending(x => x)
-            .First();
+        var periodScheme = await GetEffectivePeriodSchemeAsync(targetDate, asset, category);
+        var defaultAvailable = await GetEffectivePeriodUsableAsync(asset, category) is PeriodUsable.Accept;
+        var timeInAdvance = await GetEffectiveTimeInAdvanceAsync(asset, category);
 
-        var endingDateTime = searchDate.Add(endingTime);
+        var models = periodScheme.Periods.Select(x => new PeriodOccupancyModel(targetDate, x.StartingTime,
+            x.GetEndingTime(), periodScheme.Id, x.Id, defaultAvailable)).ToList();
 
-        var appliedAssetSchedules =
-            await _assetScheduleRepository.GetAssetScheduleListInScopeAsync(asset.Id, searchDate, endingDateTime);
+        var schedules = await _assetScheduleRepository.GetListAsync(targetDate, asset.Id, periodScheme.Id);
+        var occupancies = await _repository.GetListAsync(targetDate, asset.Id);
 
-        var effectiveSchedules =
-            await CalculateEffectiveSchedulesAsync(searchDate, endingDateTime, appliedAssetSchedules, asset, category);
+        UpdatePeriodsUsableBySchedules(models, schedules);
+        UpdatePeriodsUsableByOccupancies(models, occupancies);
```

Comment:
we should calculate all **available** period before `UpdatePeriodsUsableByOccupancies`.